src/System/IO/Resource.hs

```diff
@@ -1,3 +1,4 @@
+{-# OPTIONS_GHC -fno-warn-name-shadowing #-}
```

Comment:
Why?

Reply:
Because of `RebindableSyntax` shenanigans.

Reply:
Can you clarify? At any rate - add a comment. When `RebindableSyntax` goes away we'll remember to reenable this warning that way.